### PR TITLE
devices/bitbox02: add taproot keypath spending support for bitbox02

### PR DESCRIPTION
**Pull request by Beerosagos**

### CODE REVIEW

**Review by benma — APPROVED** [review]
cc @achow101 - we tested this PR internally before submitting it

File: hwilib/devices/bitbox02.py [hunk 1]
```diff
@@ -732,7 +746,19 @@ def script_config_from_utxo(
         for output_index, psbt_out in builtins.enumerate(psbt.outputs):
             tx_out = psbt_out.get_txout()
 
-            _, keypath = find_our_key(psbt_out.hd_keypaths)
```
Comment (benma) [inline hwilib/devices/bitbox02.py:735]: on line 721 above, prevtx can be None, so should be:

```
"prev_tx": None if prevtx is None else { .... }
```

Comment (Beerosagos) [reply hwilib/devices/bitbox02.py:735]: Thx, updated it!

**Review by benma — APPROVED** [review]

**Comment by benma** [comment]
ping @achow101 @instagibbs - would be great to include this in the next release, as we have users asking for this feature.

File: hwilib/devices/bitbox02.py [hunk 2]
```diff
@@ -673,12 +679,20 @@ def script_config_from_utxo(
                 utxo = psbt_in.witness_utxo
             if utxo is None:
                 raise BadArgumentError("No utxo found for input {}".format(input_index))
-            if prevtx is None:
-                raise BadArgumentError(
-                    "Previous transaction missing for input {}".format(input_index)
-                )
```
Comment (achow101) [inline hwilib/devices/bitbox02.py:679]: Shouldn't this check be retained for non-taproot scripts?

File: hwilib/devices/bitbox02.py [hunk 3]
```diff
@@ -673,12 +679,20 @@ def script_config_from_utxo(
...
+            key_origin_infos = psbt_in.hd_keypaths.copy()
+            if len(psbt_in.tap_internal_key) > 0:
+                # adding taproot keys to the keypaths to be checked
+                for pubkey, (leaf_hashes, key_origin_info) in psbt_in.tap_bip32_paths.items():
+                    if len(leaf_hashes) > 0:
+                        raise BadArgumentError(
+                            "The BitBox02 does not support Taproot script path spending. Found leaf hashes: {}"
+                            .format(leaf_hashes)
+                        )
+                    key_origin_infos.update({pubkey: key_origin_info})
```
Comment (achow101) [inline hwilib/devices/bitbox02.py:692]:
```suggestion
                    key_origin_infos[pubkey] = key_origin_info
```

File: hwilib/devices/bitbox02.py (hunk 2 above)
Comment (Beerosagos) [reply hwilib/devices/bitbox02.py:679]: We decided to remove it, since it is already done on the device

File: hwilib/devices/bitbox02.py (hunk 3 above)
Comment (Beerosagos) [reply hwilib/devices/bitbox02.py]: Good point, thx! :+1:

**Comment by benma** [comment]
bump @achow101

**Comment by achow101** [comment]
Did not test, Code Review ACK 59379b30a54b3009b0e8bfbbddcfc0fdf46c9d1a

**Comment by Beerosagos** [comment]
Thank you!

**Pull request merged**